### DISCUSSION TITLE
Design tips needed

### DISCUSSION THREAD

**Discussion by ycaoT** [twilio-labs/paste#1628] [Office hours] [answered]
I have a UI to build json, which includes different type of states. In my code, my main component is a state container and each state according to its type. 

There is one special type of state, it can create sub-states(same as all other types of state). What is a good way to implement this state(possibly not to create too much duplicated codes)? This state itself is like a state container, but have different logic to handle actions.    

The component structure will be:
![image](https://user-images.githubusercontent.com/61073239/124161737-640b1f00-da63-11eb-8f02-61564278f030.png)

**Comment by nkrantz** [comment]
Thanks for creating this discussion, @ycaoT! We'll review before Office Hours today.

Adding the link to your sample app here for reference: https://code.hq.twilio.com/pages/sec-eng/socless-web/

**Reply by ycaoT** [reply]
@nkrantz  can you close this one?

**Comment by serifluous** [comment, marked as answer]
Recording for this office hours session: https://www.loom.com/share/3006beed70e34f879d380ae1fd9ba4ea

Password can be found in Slack. Thanks for coming!
